src/app/compressed-air-inventory/compressed-air-inventory-setup/end-uses-table/end-uses-table.component.ts

```diff
@@ -23,29 +24,49 @@ export class EndUsesTableComponent implements OnInit {
   deleteSelectedId: string;
   confirmDeleteData: ConfirmDeleteData;
 
+  selectedSystemIdSub: Subscription;
+  selectedSystemId: string;
+  selectedSystemIndex: number;
```

Comment:
I can't see a reason why we need to track selectedSystemIndex here. We already have the systemId from the .behaviorSubject. We can use that to get the system, check that it exists and perform operations. Indexing into an array and chaining another operation (particularly inside of an array property of that object is more error prone).